src/router.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { Outlet, createBrowserRouter, redirect } from 'react-router-dom';
+import App from './App';
+import NotFound from './pages/NotFound';
+
+const jwt = localStorage.getItem('access_token');
+
+const privateMiddleware = () => {
+  if (jwt) {
+    return true;
+  }
+
+  return redirect('/signin');
+};
+
+const publicMiddleware = () => {
+  if (jwt) {
+    return redirect('/todo');
+  }
+
+  return true;
+};
+
+const route = [
+  {
+    element: <App />,
+    errorElement: <NotFound />,
+    children: [
+      {
+        path: '/todo',
+        element: (
+          <div>
+            <h1>Header</h1>
+            <Outlet />
+          </div>
+        ),
+        loader: privateMiddleware,
+        children: [
+          {
+            index: true,
+            element: <div>todo</div>,
+          },
+        ],
+      },
```

Comment:
개인적인 의견으로는 Todo 페이지에 header 넣고 라우터에서는 Todo 페이지로 이동시키는 게 가독성에는 더 좋은 것 같습니다..!

Reply:
저도 동의해요. Header를 붙이는 로직은 TodoPage 내에 있는 게
역할 분리가 확실히 될 것 같아요!

Reply:
Header와 같은 부분들은 특정 페이지의 기능과 관계없이 반복되는 레이아웃 입니다.
Outlet이 v6에 들어온 뒤로 이런 공통 레이아웃을 Page단으로부터 분리할 수 있게 되었습니다.

The <Outlet element is used as a placeholder. In this case an <Outlet>enables the Users component to render its child routes. Thus the<Outlet> element will render either a <UserProfile> or <OwnUserProfile> element depending on the current location.

Thus the best way this element can be used is in layouts. You can simply create multiple layouts, for e.g.- the Dashboard Layout(for the profile and members page) or the Main Layout(for login and logout, basically when the user is not signed in). Finally, you can simply inject whichever component one needs depending on its route rather than wrapping each route component in a parent layout component like: 

Outlet의 존재의의는 레이아웃을 페이지단으로부터 분리할 수 있고 + 여러개의 레이아웃을 만들어서 라우트단에서 미리 분리 적용 가능하다는 점 입니다. Outlet을 잘 쓰려면 Header와 같은 레이아웃을 Outlet과 함께 분리하여야합니다.

저희는 Todo페이지에서만 헤더를 사용하므로 페이지에 넘겨도 상관없겠으나 현재 app.jsx의 Outlet은 어떠한 역할을 하고 있지 않는 것도 사실입니다.

보통 이런 멀티 레이아웃을 만들때는 `const route = []` 의 라우트를 두개를 만들고(헤더 존재하는 라우트, 없는 라우트) 두개를 한꺼번에 `createBrowserRouter`에 넘기는 식으로 작업합니다. App.jsx에서는 두가지의 레이아웃을 생성해놔야 합니다. 한가지는

```javascript
<Header/>
<Outlet/>
```
의 레이아웃이고 나머지 하나는

```javascript
<Outlet/>
```
의 레이아웃 입니다.

App.jsx에서 두가지의 레이아웃을 생성 후 , router.jsx에서 두가지를 import합니다. 두 레이아웃을 각각 element에 넣은 후, 각 라우팅에 맞게 children을 짜면 됩니다.

어렵거나 이해가 안가면 말씀주세요. 다른 분들 말씀했듯이 Header를 Page로 넘겨도 되지만 Route입장에서 좋은 연습은 아닙니다.

[Outlet](https://medium.com/age-of-awareness/amazing-new-stuff-in-react-router-v6-895ba3fab6af)

Reply:
global header, global footer 같이 모든 곳에서 쓰이는 요소가 없는 지금 같은 상황에서는 Outlet보다는 App.jsx를 지우고 경로 이동하는 것이 좋아봅니다! Outlet은 나중에 global 요소가 있는 상황에서 사용하는 것이 좋을 것 같습니다..!

Reply:
네 부근님이 잘 판단하셔서 작업해주세요! 어떤 것도 괜찮습니다